app/models/submission.rb

```diff
@@ -71,8 +75,8 @@ def selected_to_advance?
     winner?
   end
 
-  def eligibility_checkbox_disabled?
-    evaluators.any?
+  def eligibility_deselection_disabled?
+    evaluator_submission_assignments.exists?(status: [:assigned, :recused])
   end
 
   def advancement_checkbox_disabled?
```

Comment:
for model method names I prefer to avoid references to UI state (e.g. disabled) and refer to the state of model instead such as:
```suggestion
  def evaluators_assigned?
    evaluator_submission_assignments.exists?(status: [:assigned, :recused])
  end

  def evaluations_missing_or_incomplete?
```

Reply:
I think you can fix the Bullet Unoptimized query "USE eager loading detected" by [changing this line](https://github.com/GSA/Challenge_platform/blob/bde972de60cc063c08db949de8f440b6f0d3e2b3/app/models/submission.rb#L83):
https://github.com/GSA/Challenge_platform/blob/bde972de60cc063c08db949de8f440b6f0d3e2b3/app/models/submission.rb#L83
by removing the reference to `evaluations` and using the assignment assocation instead:
```ruby
!eligible_for_evaluation? || !all_evaluations_completed? || evaluator_submission_assignments.empty?
```